kof-operator/internal/controller/clusterdeployment_kof_cluster_regional_role.go

```diff
@@ -2,668 +2,171 @@ package controller
-			return fmt.Errorf("failed to update config map: %v", err)
+	if cm == nil {
+		if err := r.CreateConfigMap(configData); err != nil {
+			return fmt.Errorf("failed to create ConfigMap: %v", err)
 		}
+		return nil
 	}
 
-	return nil
-}
-
-func (r *RegionalClusterRole) GetChildClusters() ([]*ChildClusterRole, error) {
-	regionalCloud := getCloud(r.clusterDeployment)
-	childClusterRoleList := make([]*ChildClusterRole, 0)
-
-	childClusterDeploymentsList := &kcmv1beta1.ClusterDeploymentList{}
-	opts := []client.ListOption{client.MatchingLabels{KofClusterRoleLabel: "child"}}
-
-	if err := r.client.List(r.ctx, childClusterDeploymentsList, opts...); err != nil {
-		return nil, fmt.Errorf("failed to get ClusterDeployments list: %v", err)
+	if err := r.UpdateConfigMap(cm, configData); err != nil {
+		return fmt.Errorf("failed to update ConfigMap: %v", err)
 	}
 
-	for _, childClusterDeployment := range childClusterDeploymentsList.Items {
-		if regionalCloud != getCloud(&childClusterDeployment) {
-			continue
-		}
-
-		childClusterDeploymentConfig, err := ReadClusterDeploymentConfig(
-			childClusterDeployment.Spec.Config.Raw,
-		)
-		if err != nil {
-			continue
-		}
-
-		if locationIsTheSame(
-			regionalCloud,
-			r.clusterDeploymentConfig,
-			childClusterDeploymentConfig,
-		) && (childClusterDeployment.Labels[KofRegionalClusterNameLabel] == "" ||
-			childClusterDeployment.Labels[KofRegionalClusterNameLabel] == r.clusterDeployment.Name) {
-			childClusterRole, err := NewChildClusterRole(r.ctx, &childClusterDeployment, r.client)
-			if err != nil {
-				return nil, fmt.Errorf("failed to create child cluster: %v", err)
-			}
-
-			childClusterRoleList = append(childClusterRoleList, childClusterRole)
-		}
-	}
-	return childClusterRoleList, nil
-}
-
-// To migrate PromxyServerGroup from releaseNamespace to clusterNamespace for multi-tenancy,
-// we need to delete the old PromxyServerGroup in the releaseNamespace first.
-func (r *RegionalClusterRole) DeleteOldPromxyServerGroup() error {
-	promxyServerGroup := &kofv1beta1.PromxyServerGroup{}
-	if err := r.client.Get(r.ctx, types.NamespacedName{
-		Name:      r.GetPromxyServerGroupName(),
-		Namespace: r.releaseNamespace,
-	}, promxyServerGroup); err != nil {
-		if errors.IsNotFound(err) {
-			return nil
-		}
-		return err
-	}
-	if err := r.client.Delete(r.ctx, promxyServerGroup); err != nil {
-		return err
-	}
 	return nil
 }
 
-func (r *RegionalClusterRole) GetPromxyServerGroup() (*kofv1beta1.PromxyServerGroup, error) {
-	promxyServerGroup := &kofv1beta1.PromxyServerGroup{}
+func (r *RegionalClusterRole) GetConfigMap() (*corev1.ConfigMap, error) {
+	configMap := &corev1.ConfigMap{}
 	if err := r.client.Get(r.ctx, types.NamespacedName{
-		Name:      r.GetPromxyServerGroupName(),
-		Namespace: r.clusterNamespace,
-	}, promxyServerGroup); err != nil {
+		Name:      GetRegionalClusterConfigMapName(r.clusterName),
+		Namespace: r.clusterDeployment.Namespace,
```

Comment:
We have `r.clusterNamespace` already:
```suggestion
		Namespace: r.clusterNamespace,
```

Reply:
* We do have it in kof (origin) `main`: https://github.com/k0rdent/kof/blob/eb2f283e26c848c7cbffd20e9ae40226d0bae237/kof-operator/internal/controller/clusterdeployment_kof_cluster_regional_role.go#L26-L29
* but we don't have it in your branch: https://github.com/gmlexx/kof/blob/afbdc48c3dcb411bff3aa1623c55c7112a192ab6/kof-operator/internal/controller/clusterdeployment_kof_cluster_regional_role.go#L21-L23
* If possible, please update your branch from `main`.